js/defaults_dialog.js

```diff
@@ -751,17 +905,60 @@ helper.defaultsDialog = (function () {
 
                     mspHelper.sendServoMixer(function () {
                         mspHelper.sendMotorMixer(function () {
-                            privateScope.finalize(selectedDefaultPreset);
+                            MSP.send_message(MSPCodes.MSP_SELECT_SETTING, [currentProfile], false, privateScope.finalize(selectedDefaultPreset));
                         })
                     });
                 } else {
-                    privateScope.finalize(selectedDefaultPreset);
+                    MSP.send_message(MSPCodes.MSP_SELECT_SETTING, [currentProfile], false, privateScope.finalize(selectedDefaultPreset));
                 }
+            })
+        });
+    };
 
-                
+    publicScope.resetSettings = function () {
+        mspHelper.getSetting("applied_defaults").then(function(aD) {
+            let selectedDefaultPreset = privateScope.getDefaults(aD.value);
 
-            })
+            if (selectedDefaultPreset && selectedDefaultPreset.settings) {
+                Promise.mapSeries(selectedDefaultPreset.settings, function (input) {
+                    return mspHelper.getSetting(input.key);
+                }).then(function () {
+                    Promise.mapSeries(selectedDefaultPreset.settings, function (input) {
+                        if (input.allow_reset) {
+                            return mspHelper.setSetting(input.key, input.value);
+                        }
+                    }).then(setTimeout( function() { 
+                        privateScope.finalize(selectedDefaultPreset);
+                    }, 6000));
+                });
+            }
+        }); 
+    };
+
+    privateScope.setSetting = function (key, value) {
+        MSP.send_message(MSPCodes.MSP_SELECT_SETTING, [0], false, function () {
+            mspHelper.setSetting(key, value, function() {
+                MSP.send_message(MSPCodes.MSP_SELECT_SETTING, [1], false, function () {
+                    mspHelper.setSetting(key, value, function() {
+                        MSP.send_message(MSPCodes.MSP_SELECT_SETTING, [2], false, function () {
+                            mspHelper.setSetting(key, value);
```

Comment:
But why 3 times? This causes everything to load 3 times and finally does not work. I don't get it.